src/lib.rs

```diff
@@ -227,11 +227,11 @@ macro_rules! create_init_submodule {
             $(
                 let modules = _py.import("sys")?.getattr("modules")?;
                 $(
-                let submod = $crate::pyo3::types::PyModule::new(_py, &$mod_name)?;
                 let qualified_name = format!("{}.{}", _name, $mod_name);
+                let submod = $crate::pyo3::types::PyModule::new(_py, &qualified_name)?;
                 $init_submod(&qualified_name, _py, submod)?;
-                modules.set_item(qualified_name, submod)?;
-                m.add_submodule(submod)?;
+                m.add($mod_name, submod)?;
+                modules.set_item(&qualified_name, submod)?;
```

Comment:
Is there a specific reason these lines got swapped?

Reply:
Good eye. Ordering isn't important here, just a consequence of me iterating over different permutations of things before landing on this change.